stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/mediapreview/MediaGalleryPreviewViewModel.kt

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.viewmodel.mediapreview
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.models.ConnectionState
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.setup.state.ClientState
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * A ViewModel capable of loading images, playing videos.
+ */
+public class MediaGalleryPreviewViewModel(
+    private val chatClient: ChatClient,
+    private val clientState: ClientState,
+    private val messageId: String,
+) : ViewModel() {
+
+    /**
+     * The currently logged in user.
+     */
+    public val user: StateFlow<User?> = chatClient.clientState.user
+
+    /**
+     * Indicates if we have fetched the complete message from the backend.
+     *
+     * This is necessary because our first state is set via a minimum viable
+     * data set needed to display the full UI in offline state.
+     *
+     * @see [io.getstream.chat.android.compose.state.mediagallerypreview.MediaGalleryPreviewActivityAttachmentState]
+     * and [io.getstream.chat.android.compose.ui.attachments.preview.MediaGalleryPreviewActivity.getIntent]
+     */
+    internal var hasCompleteMessage: Boolean = false
```

Comment:
If we're offline we load the first set of data as an "incomplete data" through parcelization.
Said data is enough to perform most gallery actions, however we still want to fetch a complete message so we track our efforts with this boolean.